### PR TITLE
Fix investment project test flakiness

### PR DESCRIPTION
**Pull request by oliverjwroberts**
## Description of change

Several investment project tests were identified as flaky and this PR attempts to fix them.

### project-edit-details-spec

```
- When changing the project FDI type to Capital only
  - should take the user to a confirmation step and display the warning
    - AssertionError: Timed out retrying after 5000ms: Expected <div#field-fdi_type.src__FormGroup-sc-1m4431t-0.sc-bBeLUv.gMksRW.cFOgVy> not to exist in the DOM, but it was continuously found.
  - should submit the request with zero new and safeguarded jobs, and null for average salary
    - AssertionError: Timed out retrying after 5000ms: Expected to find content: 'Submit' within the selector: 'button' but never did.
- When changing the project FDI type from Expansion to other
  - should submit the request with investor type set to the user selection:
    - CypressError: Timed out retrying after 5000ms: `cy.wait()` timed out waiting `5000ms` for the 1st request to the route: `editDetailsRequest`. No request ever occurred.
```
These have been fixed by adding assertions after button actions to ensure the DOM has loaded before the Cypress command resolves and moves onto the next.

### project-edit-value-spec

```
- When viewing a capital intensive project with no value fields set
  - should render the header
    - CypressError: `cy.contains()` cannot be passed an empty string.
- When viewing a capital intensive project with no value fields set
  - should render the header
    - CypressError: `cy.contains()` cannot be passed an empty string.
- When viewing a capital intensive project with no value fields set
  - should render breadcrumbs:
    - CypressError: `cy.contains()` cannot be passed an empty string.
```

These have been fixed by setting the `name` property to a non-empty string in the `investmentProjectFaker`. Previously, the schema was occasionally generating an empty string causing `cy.contains(project.name)` to raise an error.

## Test instructions

The following specs should now consistently pass.
- project-edit-details-spec
- project-edit-value-spec

## Checklist

[//]: # 'When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/main/docs/Code%20review%20guidelines.md'

- [x] Has the branch been rebased to main?
- [x] Automated tests (Any of the following when applicable: Unit, Functional or End-to-End)
- [x] Manual compatibility testing (Browsers: Chrome, Firefox, Edge, Safari)

### CODE REVIEW

**Review by dredmonds — APPROVED** [review]

File: test/functional/cypress/specs/investments/project-edit-details-spec.js [hunk 1]
```diff
@@ -66,6 +66,7 @@ const setup = (project) => {
   }).as('getProjectDetails')
   cy.visit(urls.investments.projects.editDetails(project.id))
   cy.wait('@getProjectDetails')
+  cy.get('[data-test="submit"]').should('be.visible')
```
Comment (peterhudec) [inline test/functional/cypress/specs/investments/project-edit-details-spec.js:69]: Are you sure that changes to this spec file are needed? It seems that they are not related to the missing `name`.

Comment (oliverjwroberts) [reply test/functional/cypress/specs/investments/project-edit-details-spec.js:69]: Changes to this file ensure the DOM is loaded before Cypress tries to carry out other assertions. The `name` property was causing issues in the `project-edit-value-spec` instead.

**Review by peterhudec — APPROVED** [review]
